src/main/java/com/conversantmedia/util/concurrent/PushPullBlockingQueue.java

```diff
@@ -299,31 +299,24 @@ public boolean addAll(Collection<? extends E> c) {
 
     @Override
     public boolean removeAll(Collection<?> c) {
-        int numFalses = 0;
-        for (final Object o : c) {
-            if (!remove(o)) numFalses++;
-        }
-        return numFalses > 0;
+         c.forEach(this::remove);
```

Comment:
This syntax is only available in Java 8.    I wouldn't mind as much if the implementation is correct but its something to consider.